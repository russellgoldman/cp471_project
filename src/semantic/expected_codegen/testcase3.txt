number = 0
i = 0
L1: if i < 10 goto L2
goto L3
L2: number = number + i
number = 3
number = 5
  i = i + 1
goto L1
L3: