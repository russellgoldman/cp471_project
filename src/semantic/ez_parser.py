# Ref: https://ply.readthedocs.io/en/latest/ply.html
import ply.yacc as yacc
import grammar_rules_ast
from lexer import Lexer

class Parser(object):
    # Build the parser using the grammar rules module
    def build(self, **kwargs):
        self.lexer = Lexer()
        self.lexer.build()
        self.parser = yacc.yacc(module=grammar_rules_ast, **kwargs)

    def parse(self, data, debug=False):
        return self.parser.parse(data, debug=debug, lexer = self.lexer)