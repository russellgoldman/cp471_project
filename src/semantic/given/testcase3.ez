Number number = 0
for(Number i = 0; i < 10; i++) {
    number = number + i
}
