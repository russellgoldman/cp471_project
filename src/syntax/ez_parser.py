# Ref: https://ply.readthedocs.io/en/latest/ply.html
import sys, os

sys.path.append(os.path.abspath(os.path.join('..')))
import ply.yacc as yacc
import grammar_rules
from lexical.lexer import Lexer

class Parser(object):
    # Build the parser using the grammar rules module
    def build(self, **kwargs):
        self.lexer = Lexer()
        self.lexer.build()
        self.parser = yacc.yacc(module=grammar_rules, **kwargs)

    def parse(self, data, debug=False):
        return self.parser.parse(data, debug=debug, lexer = self.lexer)