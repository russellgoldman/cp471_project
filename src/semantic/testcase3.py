import unittest
import sys, os
sys.path.append(os.path.abspath(os.path.join('..', 'utilities')))

from ez_parser import Parser

class TestCase2(unittest.TestCase):
    def test_for_loop(self) -> None:
        source_f = open('./given/testcase3.ez', 'r')
        source = source_f.read()
        source_f.close()

        expected_f = open('./expected/testcase3.txt', 'r')
        expected = expected_f.read()
        expected_f.close()

        # build parser
        p = Parser()
        p.build()
        # generate parse tree
        given_tree = p.parse(source)
        print(given_tree)
        self.assertEqual(str(given_tree), expected)

if __name__ == '__main__':
    unittest.main()