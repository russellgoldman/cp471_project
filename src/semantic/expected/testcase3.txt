('statement', ('assignmentExpression', '=', 'number', 0), ('forStatement', ('forExpression', ('assignmentExpression', '=', 'i', 0), ('forExpression', ('relationExpression', '<', 'i', 10), ('assignmentExpression', 'i', '=', ('iterationExpression', '+', 'i', 1)))), ('assignmentExpression', '=', 'number', ('sumExpression', '+', 'number', 'i'))))